### ISSUE TITLE
jacoco-sonarqube-incompatible-version-1007

### ISSUE THREAD

**Issue by JarlSX** [jacoco/jacoco#315]
Jacoco version: 0.7.5.201505241946 triggers the following error: 
java.io.IOException: Incompatible version 1007

see also: http://stackoverflow.com/questions/30459260/jacoco-sonarqube-incompatible-version-1007

**Comment by benzonico** [comment]
This is more an issue on SonarQube side : http://jira.codehaus.org/browse/SONARJAVA-1091

**Comment by Godin** [comment]
This is not a bug on JaCoCo side, but intentional change for binary dump file as consequence of #310 - see changelog http://www.eclemma.org/jacoco/trunk/doc/changes.html
Third-party integrations, which use this binary dump file (such as SonarQube) should be updated, or previous version of JaCoCo should be used for them.

**Comment by Godin** [comment]
@marchof what do you think about feasibility of support reading of several different binary versions for analyzer?

**Comment by marchof** [comment]
@Godin Typically incompatibility come from the fact that we change the strategy how to insert probes. To read different versions at the same time we would need to keep copies of previous ProbeAdapters and Analyzers. In general I don't consider exec files as a "persitence format", it's a short term storage to communicate between the agent and the analyzer. I would rather see the XML reports as official and stable API.

**Comment by laba69** [comment]
Jacoco last version I installed :jacoco-0.7.6-20150603.214955-4
triggers the same error in eclipse:
Error while importing coverage session.
Reason: Error while reading execution data file : C:/tmp/jacoco.exec(code 5005)
in the details view I found: Incompatible version 1007.

I installed EclEmma jacoco plugin today.
This error is about the plugin or about my jacoco exec file?
How can i solve this properly? I'm not comfortable with maven.
Thank you for your contribution
